lib/ansiblelint/rules/PlaybookExtension.py

```diff
@@ -2,22 +2,25 @@
 # Copyright (c) 2018, Ansible Project
 
 import os
-from typing import List
+from typing import TYPE_CHECKING, List, Optional, Union
 
 from ansiblelint.rules import AnsibleLintRule
 
+if TYPE_CHECKING:
+    from ansiblelint.file_utils import TargetFile
+
 
 class PlaybookExtension(AnsibleLintRule):
     id = '205'
     shortdesc = 'Use ".yml" or ".yaml" playbook extension'
     description = 'Playbooks should have the ".yml" or ".yaml" extension'
     severity = 'MEDIUM'
     tags = ['formatting']
-    done = []  # type: List  # already noticed path list
+    done: List[str] = []
     version_added = 'v4.0.0'
 
-    def match(self, file, text):
-        if file['type'] != 'playbook':
+    def match(self, file: Optional["TargetFile"] = None, line: str = "") -> Union[bool, str]:
+        if not file or file['type'] != 'playbook':
```

Comment:
This feels like an invasion into the logic that is out of the scope of this PR. Could you show places where `file` is passed as `None`?